### PR TITLE
Fix softmodules

### PR DESCRIPTION
**Pull request by LucMc**
## Description
Updated the training loop in the `softmodules_metaworld_jax.py` code to support the latest gymnasium version. Previously it would throw an error on line 687 where "final_info" was no longer part of the information returned after an env step. This was done by combining the old training loop with the one in `mtmhsac_jax_checkpoint.py`. I've tested the change on MT10 and got expected loss curves. Furthermore there were a number of minor changes given by the pre-commit. The openrlbenchmark was throwing an index error for me.
## Types of changes

- [X] Bug fix
- [ ] New feature
- [ ] New algorithm
- [ ] Documentation

## Checklist:

- [X] I've read the [CONTRIBUTION](https://docs.cleanrl.dev/contribution/) guide (**required**).
- [X] I have ensured `pre-commit run --all-files` passes (**required**).
- [ ] I have updated the tests accordingly (if applicable).
- [ ] I have updated the documentation and previewed the changes via `mkdocs serve`.
    - [ ] I have explained note-worthy implementation details.
    - [ ] I have explained the logged metrics.
    - [ ] I have added links to the original paper and related papers.

If you need to run benchmark experiments for a performance-impacting changes:

- [ ] I have contacted @vwxyzjn to obtain access to the [openrlbenchmark W&B team](https://wandb.ai/openrlbenchmark).
- [ ] I have used the [benchmark utility](/get-started/benchmark-utility/) to submit the tracked experiments to the [openrlbenchmark/cleanrl](https://wandb.ai/openrlbenchmark/cleanrl) W&B project, optionally with `--capture-video`.
- [ ] I have performed RLops with `python -m openrlbenchmark.rlops`.
    - For new feature or bug fix:
        - [ ] I have used the RLops utility to understand the performance impact of the changes and confirmed there is no regression.
    - For new algorithm:
        - [ ] I have created a table comparing my results against those from reputable sources (i.e., the original paper or other reference implementation).
    - [ ] I have added the learning curves generated by the `python -m openrlbenchmark.rlops` utility to the documentation.
    - [ ] I have added links to the tracked experiments in W&B, generated by `python -m openrlbenchmark.rlops ....your_args... --report`,  to the documentation.

### CODE REVIEW

**Comment by rainx0r** [comment]
I think you might need to update the `Agent` class' checkpoint related code but other than that it seems fine

**Comment by rainx0r** [comment]
There's also missing imports (e.g. `load_env_checkpoints`)
